### PR TITLE
Update to React 16

### PR DESCRIPTION
**Pull request by nishacodes**
**Summary**
Updated the datepicker to use React 16. In doing so, updated peer dependencies and the new style of defining PropTypes.

Also created a new option to make the input field for dates and datetimes editable. The way this works is you set the `inputEditable` prop to true. Whatever you type in the field will be set as the date when you hit enter OR when onBlur (i.e. when you click out of the field). 

**Test Plan**
- [ ] Pull down branch and run `yarn && yarn build`
- [ ] `open index.html` and test out the various date pickers on the demo page.
- [ ] test the input editable demo as well -- try typing various date and time formats

### CODE REVIEW

**Comment by georgemayer** [comment]
This works pretty great! There are obviously unsupported date formats, but it seems pretty clear base on the format that's in there. Also, technically you can toggle on both editable and time, but time isn't editable. I'm not sure if we care to support that. A few things: 

- [ ] I think maybe we want to remove the outline on focus when it's not editable
- [ ] perhaps we want to try and use this package to parse what people put in. If it's anything like the ruby package, it's pretty powerful: https://github.com/wanasit/chrono

**Comment by georgemayer** [comment]
```

File: app/js/date-view.jsx [hunk 1]
```diff
@@ -259,4 +251,11 @@ class DatePicker extends React.Component {
     }
 }
 
+DatePicker.propTypes = {
```
Comment (georgemayer) [inline app/js/date-view.jsx]: Do we have to move these out of the component?

File: app/js/datepicker.jsx [hunk 2]
```diff
@@ -189,6 +197,58 @@ class DatePicker extends React.Component {
...
+    }
+
+    handleEndDateInputChange(e) {
+        this.setState({
+            endDateInputValue: e.target.value
+        });
+    }
+
+    handleStartDateKeyPress(e) {
+        if (e.charCode === CHAR_CODE_ENTER) {
```
Comment (georgemayer) [inline app/js/datepicker.jsx]: this is a deprecated event field: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode

**Comment by georgemayer** [comment]
ok, I pushed up one commit to reset the datepicker to show the date you pick on enter, but I think we also want a few other things:
- [ ] to keep the datepicker open when you're editing
- [ ] to reformat the date for the input when you hit enter

**Comment by nishacodes** [comment]
Those suggestions make sense 👍. I'll have a look!

**Comment by georgemayer** [comment]
Awesome work!

**Pull request merged**